schema_editor/app/scripts/schemas/schemas-service.js

```diff
@@ -64,7 +96,7 @@
                             },
                             enumSource: [{
                                     source: 'target',
-                                    title: fieldData.referenceTarget + ' {{i}}',
+                                    title: displayProperties.join(' '),
```

Comment:
What do you think about a max length for this text? Maybe on a per-property basis too: in case some paragraph text (like a description), or a long street address is involved.